lib/swagger-scanner.ts

```diff
@@ -170,4 +193,71 @@ export class SwaggerScanner {
     );
     return modulePath ?? Reflect.getMetadata(MODULE_PATH, metatype);
   }
+
+  private scanModuleImportsRecursively(
+    imports: Set<Module>,
+    container: NestContainer,
+    currentDepth: number,
+    maxDepth: number | undefined,
+    processedModules: Set<string>,
+    options: {
+      globalPrefix: string;
+      internalConfigRef: ApplicationConfig;
+      operationIdFactory?: OperationIdFactory;
+      linkNameFactory?: (
+        controllerKey: string,
+        methodKey: string,
+        fieldKey: string
+      ) => string;
+      autoTagControllers?: boolean;
+    }
+  ): ModuleRoute[] {
+    let result: ModuleRoute[] = [];
+
+    for (const { metatype, controllers, imports: subImports } of imports.values()) {
+      // Skip if module has already been processed
+      const moduleId = this.getModuleId(metatype);
+      if (processedModules.has(moduleId) || container.isGlobalModule(metatype) || (maxDepth !== undefined && currentDepth > maxDepth)) {
```

Comment:
```suggestion
    for (const { token, metatype, controllers, imports: subImports } of imports.values()) {
      // Skip if module has already been processed
      if (processedModules.has(token) || container.isGlobalModule(metatype) || (maxDepth !== undefined && currentDepth > maxDepth)) {
```
Here's an example implementation of the approach I suggested